apps/advisor/pkg/app/utils.go

```diff
@@ -76,6 +77,71 @@ func processCheck(ctx context.Context, client resource.Client, obj resource.Obje
 	}, resource.PatchOptions{}, obj)
 }
 
+func processCheckRetry(ctx context.Context, client resource.Client, obj resource.Object, check checks.Check) error {
+	status := checks.GetStatusAnnotation(obj)
+	if status == "" || status == checks.StatusAnnotationError {
+		// Check not processed yet or errored
+		return nil
+	}
+	retry := checks.GetRetryAnnotation(obj)
```

Comment:
Nit: Maybe this should have a better name as it wasn't clear to me that this would be the same as the item ID 🤔